src/plugins/dashboard/public/locator.ts

```diff
@@ -7,14 +7,22 @@
  */
 
 import type { SerializableRecord } from '@kbn/utility-types';
+import { flow } from 'lodash';
 import type { TimeRange, Filter, Query, QueryState, RefreshInterval } from '../../data/public';
 import type { LocatorDefinition, LocatorPublic } from '../../share/public';
 import type { SavedDashboardPanel } from '../common/types';
+import type { DashboardPanelMap } from './types';
+import { convertSavedDashboardPanelToPanelState } from '../common/embeddable/embeddable_saved_object_converters';
 import { esFilters } from '../../data/public';
 import { setStateToKbnUrl } from '../../kibana_utils/public';
 import { ViewMode } from '../../embeddable/public';
 import { DashboardConstants } from './dashboard_constants';
 
+/**
+ * Useful for ensuring that we don't pass any non-serializable values to history.push (for example, functions).
+ */
+const getSerializableRecord: <O>(o: O) => O & SerializableRecord = flow(JSON.stringify, JSON.parse);
+
```

Comment:
interface DashboardAppLocatorParams should not extend SerializableRecord, it should just make sure itself satisfies SerializableRecord constrains. (line 37), i know its not part of this PR, but this allows you to pass in more options than you should in this PR